### PR TITLE
Automated formatting

### PR DESCRIPTION
**Pull request by Andrei-Aksionov**
Hi there 👋 

Another round of code-style fixes:
```bash
ruff check * \
    --select E,W,F,S \
    --extend-select ANN001,ANN201,ANN205,ANN206,ARG001,C4,I001,PT,RET,SIM \
    --ignore C408,E402,E501,E731,PT004,PT007,S101,S108,S113,S310,S603 \
    --per-file-ignores "*/**/__init__.py":I001,"*/**/__init__.py":F401,"tests/*":ANN,"*/**/unsloth/kernels/*":ALL \
    --line-length 120 \
    --target-version py38 \
    --fix-only; \
black \
    -l120 \
    -C \
    --preview \
    --target-version py38 \
    config_hub eval extensions litgpt tests \
    --exclude unsloth/kernels
```

Notable changes:
- disabled F401 (imported but unused) for `__init__.py` files.
- excluded `unsloth/kernels` from linting.
- added `extensions` folder to the list.

---

For those preferring `pre-commit` I updated a corresponding [config](https://github.com/Lightning-AI/litgpt/issues/758#issuecomment-1862522682).

### CODE REVIEW

File: extensions/thunder/unsloth/kernels/cross_entropy_loss.py
Comment (carmocca) [inline extensions/thunder/unsloth/kernels/cross_entropy_loss.py:1]: The kernels directory shouldnt be formatted to keep the original style of the code

Comment (Andrei-Aksionov) [reply extensions/thunder/unsloth/kernels/cross_entropy_loss.py:1]: Oops.
Fixed.
\+ updated CLI command in the PR description.

**Comment by Borda** [comment]
Why not add the pre-commit or at least write this configuration to the already existing `pyproject.toml`?

**Comment by Andrei-Aksionov** [comment]
@Borda when Lit-GPT was a set of scripts it was not the option, since the decision was made to not have such config files in the repo (I believe for the sake of simplicity).
But now we have LitGPT, so I think we can revisit this question. @lantiga?

**Comment by Borda** [comment]
as the pyproject already exists, I would rather have the configuration there than in some PR description which may differ over time...

**Pull request merged**